src/framegrab/grabber.py

```diff
@@ -703,44 +684,57 @@ def _substitute_rtsp_password(self, config: dict) -> dict:
 
         return config
 
+    def _apply_camera_specific_options(self, options: dict) -> None:
+        if options.get("resolution"):
+            camera_name = self.config.get("name", "Unnamed RTSP Stream")
+            raise ValueError(f"Resolution was set for {camera_name}, but resolution cannot be set for RTSP streams.")
+
+    def _open_connection(self):
+        self.capture = cv2.VideoCapture(self.stream)
+        if not self.capture.isOpened():
+            raise ValueError(
+                f"Could not open RTSP stream: {self.stream}. Is the RTSP URL correct? Is the camera connected to the network?"
+            )
+        logger.debug(f"Initialized video capture with backend={self.capture.getBackendName()}")
+
+    def _close_connection(self):
+        with self.lock:
+            self.capture.release()
```

Comment:
This should be a little more paranoid, and check to make sure the `self.capture` object exists (is not None) before calling release on it.

Reply:
Done.